### PR TITLE
Angular front end

### PR DESCRIPTION
**Pull request by BRHoward**
- Converted front end to angular whilst retaining original functionality
- Better styling and added angular animations for entrance/exit
- Adapted tests for angular and for animations

### CODE REVIEW

**Comment by BRHoward** [comment]
@stevenhillcox @jebbinSCL @sievins

File: public/main.js [hunk 1]
```diff
@@ -1,198 +1,155 @@
...
+
+angular.module("todoList", ["ngAnimate"])
+    .controller("TodoListController", ["$scope", "$http", function($scope, $http) {
+
+        $scope.todos = [];
+        $scope.inputBoxText = "";
+        $scope.errorText = "";
+        $scope.todosLoaded = false;
+
+        $scope.getTodos = function() {
```
Comment (stevenhillcox-sl) [inline public/main.js]: As a styling suggestion, I prefer declaring functions at the bottom of the code and all scope declarations separate at the top.
Have a look at https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
Particularly the section called 'Bindable Members Up Top'
This comment applies to the rest of your code too.

File: public/main.js [hunk 2]
```diff
@@ -1,198 +1,155 @@
...
-reloadTodoList();
-var timerId = setInterval(reloadTodoList, 30000);
+        };
+
+        angular.element(document).ready(function() {
+            $scope.getTodos();
+        });
+        var pollTimer = setInterval($scope.getTodos, 30000);
+
+    }]);
```
Comment (stevenhillcox-sl) [inline public/main.js]: In general, you have a lot of logic inside your ToDoController. I would recommend pulling this out and making a ToDoService. Your controller should be a small as possible.

**Comment by BRHoward** [comment]
@stevenhillcox

**Pull request merged**